packages/site-demo/gatsby-config.js

```diff
@@ -72,5 +67,38 @@ module.exports = {
         },
         // Prevent styles from being squashed into a single CSS file.
         'gatsby-plugin-split-css',
+        // LunR search indexing
+        {
+            resolve: `./lumx-lunr-search`,
+            options: {
+                filename: 'search_index.json',
+                getDocumentsForTypes: {
+                    // Index site page with at least a title.
+                    async SitePage({ context = {} }) {
+                        const { title, slug } = context;
+                        if (!title) return;
+                        let content;
+                        if (process.env.NODE_ENV === 'production') {
+                            // In PROD, extract text from the generated page HTML.
+                            content = await extractTextFromHTMLPage(
```

Comment:
Why should we have a different approach in production and dev?

Reply:
I wanted to index all the content of the HTML pages (content generated from markdown but also from React components) and HTML pages are rendered statically only in the production build. 
In development build, the HTML is almost empty (like a normal SPA) and thus I use a content excerpt generated from the markdown by gatsby